### PR TITLE
Adding basic nox configuration and simple test.

### PR DESCRIPTION
**Pull request by dhermes**
~~**NOTE**: Uses #1 as diffbase.~~

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+"""Build and test configuration file.
+
+Assumes ``nox >= 2018.9.14`` and ``py >= 1.6.0`` are installed.
+"""
+
+import os
+import subprocess
+
+import nox
+import py.path
```
Comment (theacodes) [inline noxfile.py]: since nox is 3+ only, can you use pathlib?

File: noxfile.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+NOX_DIR = os.path.abspath(os.path.dirname(__file__))
+DEFAULT_INTERPRETER = "3.7"
+PYPY = "pypy3"
+ALL_INTERPRETERS = ("3.5", "3.5-32", "3.6", "3.6-32", "3.7", "3.7-32", PYPY)
```
Comment (theacodes) [inline noxfile.py]: I don't think we need to really test separately against 32bit interpreters.

File: noxfile.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+@nox.session(python=DEFAULT_INTERPRETER)
+def blacken(session):
+    if py.path.local.sysfind("git") is None:
+        session.skip("`git` must be installed")
+
+    # Install all dependencies.
+    session.install("black")
+    # Get all Python files to be blackened
+    file_list_str = subprocess.check_output(["git", "ls-files", "*.py"])
```
Comment (theacodes) [inline noxfile.py]: Can we just pass in `google` and `tests` to blacken instead of doing this?

File: noxfile.py (hunk 1 above)
Comment (dhermes) [reply noxfile.py]: AFAIK `pathlib` doesn't include a feature like `py.path.local.sysfind`, which is what I'm using. Maybe I'm mistaken?

File: noxfile.py (hunk 2 above)
Comment (dhermes) [reply noxfile.py]: :+1:

File: noxfile.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    return os.path.join(NOX_DIR, *names)
+
+
+@nox.session(py=ALL_INTERPRETERS)
+def unit(session):
+    # Install all dependencies.
+    session.install("pytest")
+    session.install(".")
+    # Run py.test against the unit tests.
+    run_args = ["py.test"] + session.posargs + [get_path("tests", "unit")]
```
Comment (theacodes) [inline noxfile.py]: use `"pytest"` (no dot)

File: noxfile.py (hunk 3 above)
Comment (dhermes) [reply noxfile.py]: Is there a reason not to blacken other / all files, e.g. `noxfile.py` and `docs/conf.py`? I think I'll also add [`blacken-docs`][1] as well.

[1]: https://github.com/asottile/blacken-docs

File: noxfile.py (hunk 1 above)
Comment (theacodes) [reply noxfile.py]: If you remove the usage of `git` below you won't need it anyway.

File: noxfile.py (hunk 4 above)
Comment (dhermes) [reply noxfile.py]: MERP. Thanks

File: noxfile.py (hunk 3 above)
Comment (theacodes) [reply noxfile.py]: Not really, other than it's just overkill to use git when our python files should be reasonably organized.

Comment (dhermes) [reply noxfile.py]: OK. I'll hardcode the Python files. This way the `py.path` import also goes away (for now).

**Comment by dhermes** [comment]
@theacodes PTAL, review changes have been addressed

**Pull request merged**